exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -114,356 +88,394 @@ void tearDown() {
+        MapIndex<HashCode, String> counterNames = schema.counterNames();
+        HashCode counterId = sha256().hashString(AFTER_COMMIT_COUNTER_NAME, UTF_8);
+
+        assertThat(counters.get(counterId)).isEqualTo(1L);
+        assertThat(counterNames.get(counterId)).isEqualTo(AFTER_COMMIT_COUNTER_NAME);
+        return null;
+      });
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getActualConfiguration() {
-    withNodeFake(() -> {
-      Throwable t = assertThrows(RuntimeException.class, () -> service.getActualConfiguration());
-      assertThat(t.getMessage()).contains("Couldn't not find any config for"
-          + " height 0, that means that genesis block was created incorrectly.");
-    });
+    short validatorCount = 8;
+    try (TestKit testKit = TestKit.builder(EmulatedNodeType.VALIDATOR)
+        .withValidators(validatorCount)
+        .withService(QaServiceModule.class)
+        .build()) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      StoredConfiguration configuration = service.getActualConfiguration();
+      EmulatedNode emulatedNode = testKit.getEmulatedNode();
+      PublicKey emulatedNodeServicePublicKey = emulatedNode.getServiceKeyPair().getPublicKey();
+      List<ValidatorKey> validatorKeys = configuration.validatorKeys();
+
+      assertThat(validatorKeys).hasSize(validatorCount);
+      assertThat(emulatedNodeServicePublicKey).matches(pk -> validatorKeys.stream()
+          .anyMatch(v -> v.serviceKey().equals(pk)));
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getBlockTransactionsByBlockId() {
-    withNodeFake(() -> {
-      Throwable t = assertThrows(RuntimeException.class, () -> service.getBlockTransactions(0L));
-      assertThat(t.getMessage()).contains(NO_GENESIS_BLOCK_ERROR_MESSAGE);
-    });
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
+      List<HashCode> blockTransactions = service.getBlockTransactions(1L);
+      assertThat(blockTransactions).hasSize(1);
+      assertThat(blockTransactions.get(0)).isEqualTo(createCounterTransaction.hash());
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxMessages() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionMessage> txMessages = service.getTxMessages();
-      assertThat(txMessages).isEmpty();
-    });
+      assertThat(txMessages).hasSize(2);
```

Comment:
Why does it have such size and not 1?

Reply:
Because it contains both committed and in-pool transactions and we have one committed and one in-pool (afterCommit) transaction at this moment.

Reply:
I see, thanks. I'd explain that because the only transaction in the test is one passed to `createBlockwith...`.